
syntax = "proto3";

import "google/protobuf/empty.proto";

service MKSE {

rpc setup (SetupMessage) returns (google.protobuf.Empty) {}
rpc getRofFile (getRofFileRequest) returns (getRofFileResponse) {}

rpc update (UpdateMessage) returns (google.protobuf.Empty) {}
rpc batchUpdate (BatchUpdateMessage) returns (google.protobuf.Empty) {}

rpc search (SearchMessage) returns (SearchResponse) {}

}

message SetupMessage
{
    int32 maxQueueSize = 1;
}

message getRofFileRequest
{
    int32 ind = 1;    
}

message getRofFileResponse
{  
    int32 r = 1;
}

message UpdateMessage
{
    int32 r = 1;
    bytes dcipher = 2;
    int32 ind = 3;
}

message SearchMessage
{
    bytes q = 1;    
}

message SearchResponse
{
    repeated bytes value = 1;  
    repeated int32 id = 2;
    
}

message BatchUpdateMessage
{
    repeated int32 r = 1;   
    repeated bytes dcipher = 2;   
    repeated int32 ind = 3;
}